embedded-kafka/src/test/scala/net/manub/embeddedkafka/EmbeddedKafkaWithRunningKafkaOnFoundPortSpec.scala

```diff
@@ -0,0 +1,51 @@
+package net.manub.embeddedkafka
+
+class EmbeddedKafkaWithRunningKafkaOnFoundPortSpec
+    extends EmbeddedKafkaSpecSupport
+    with EmbeddedKafka {
+
+  "the withRunningKafkaOnFoundPort method" should {
+    "start and stop Kafka and Zookeeper successfully on non-zero ports" when {
+      val userDefinedConfig = EmbeddedKafkaConfig(kafkaPort = 12345, zooKeeperPort = 12346)
+      val actualConfig = withRunningKafkaOnFoundPort(userDefinedConfig) { actualConfig =>
+        actualConfig shouldBe userDefinedConfig
+        bothKafkaAndZkAreAvailable(actualConfig)
+        actualConfig
+      }
+      bothKafkaAndZkAreNotAvailable(actualConfig)
+    }
+
+    "start and stop multiple Kafka and Zookeeper successfully on arbitrary available ports" when {
+      val userDefinedConfig = EmbeddedKafkaConfig(kafkaPort = 0, zooKeeperPort = 0)
+      val actualConfig1 = withRunningKafkaOnFoundPort(userDefinedConfig) { actualConfig1 =>
+        bothKafkaAndZkAreAvailable(actualConfig1)
+        publishStringMessageToKafka("topic", "message1")(actualConfig1)
+        consumeFirstStringMessageFrom("topic")(actualConfig1) shouldBe "message1"
+        val actualConfig2 = withRunningKafkaOnFoundPort(userDefinedConfig) { actualConfig2 =>
+          bothKafkaAndZkAreAvailable(actualConfig2)
+          publishStringMessageToKafka("topic", "message2")(actualConfig2)
+          consumeFirstStringMessageFrom("topic")(actualConfig2) shouldBe "message2"
+          val allConfigs = Seq(userDefinedConfig, actualConfig1, actualConfig2)
+          // Confirm both actual configs are running on separate non-zero ports, but otherwise equal
+          allConfigs.map(_.kafkaPort).distinct.size shouldBe 3
```

Comment:
can you please use `.distinct should have size 3`? here and wherever you use to check sizes